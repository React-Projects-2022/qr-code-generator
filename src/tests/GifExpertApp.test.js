import "@testing-library/jest-dom";
import { shallow } from "enzyme";
import { GifExpertApp } from "../GifExpertApp";

describe("Pruebas en GifExpertApp", () => {
  test("Debe mostrar que está cargando", () => {
    const wrapper = shallow(<GifExpertApp />);
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe mostrar lista de categorías", () => {
    const categories = ["React", "Angular"];
    const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("GifGrid").length).toBe(categories.length);
    expect(wrapper.find("AddCategory").length).toBe(1);
  });
});
