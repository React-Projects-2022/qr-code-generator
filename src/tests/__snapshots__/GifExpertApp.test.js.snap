// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifExpertApp Debe mostrar lista de categorías 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="React"
      key="0"
    />
    <GifGrid
      category="Angular"
      key="1"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas en GifExpertApp Debe mostrar que está cargando 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
